### PR TITLE
Tests performance improvements

### PR DESCRIPTION
**Pull request by tovbinm**
**Related issues**
Our tests became painfully slow

**Describe the proposed solution**
1. Reuse the same spark session across tests
2. Avoid setting checkpoint dirs

**Describe alternatives you've considered**
NA

** Additional context **
We need to investigate more is this really improving the performance @wsuchy

### CODE REVIEW

File: utils/src/main/scala/com/salesforce/op/test/TestSparkContext.scala [hunk 1]
```diff
@@ -60,22 +59,11 @@ trait TestSparkContext extends TempDirectoryTest with TestCommon {
...
 
-  lazy val checkpointDir: String = createDirectory(tempDir.getCanonicalPath, "checkpoints").toString
-
   override def beforeAll: Unit = {
     super[TempDirectoryTest].beforeAll()
-    sc.setCheckpointDir(checkpointDir)
+    val _ = sc // triggers initialization of spark session
   }
 
-  override def afterAll: Unit = {
```
Comment (wsuchy) [inline utils/src/main/scala/com/salesforce/op/test/TestSparkContext.scala]: Does it mean that we no longer want to cleanup temp directories created by spark?

Comment (tovbinm) [reply utils/src/main/scala/com/salesforce/op/test/TestSparkContext.scala]: It’s being cleared in the base trait

**Comment by tovbinm** [comment]
We yet have to see how much of an improvement this change is.

**Comment by tovbinm** [comment]
No improvement was observed.

**Pull request closed without merging**